### PR TITLE
Add changelog and changelog automation

### PR DESCRIPTION
**Pull request by agjohnson**
First, update the version in our setup.cfg and changelog:

    invoke prepare <version>

Then, hand edit and commit, added any release notes, etc. Tag manually or:

    invoke release <version>

Finally, push up your changes.

This:

* Uses setup.cfg instead of setup.py
* Parses setup.cfg for version
* Updates setup.cfg on release prepare task
* Show our release version in our footer
* Uses invoke for task execution
* Uses a node library to hit GitHub and fetch the changelog
* Imports this into our existing changelog

### CODE REVIEW

**Comment by ericholscher** [comment]
The description of this PR should be in the docs included in the PR ;)

**Comment by agjohnson** [comment]
We don't have a release manager doc yet, so I left this out.

**Comment by agjohnson** [comment]
Also, this updates the changelog doc with a sample of 2.0 release. This can be updated when we actually release 2.0

File: tasks.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+@task
+def prepare(ctx, version):
+    """
+    Prepare the next release version by updating files.
+
+    This will stage a few updates for manual review and commit:
+
+    * Prepend the most recent PRs and issues that were closed to CHANGELOG.md.
```
Comment (humitos) [inline tasks.py]: `CHANELOG.rst` ;)

File: tasks.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    updated.  New entries will end up at the top of the file, under a heading
+    for the new version.
+    """
+    print('Updating release version in setup.cfg')
+    setupcfg_path = os.path.join(os.path.dirname(__file__), 'setup.cfg')
+    config = RawConfigParser()
+    config.read(setupcfg_path)
+    config.set('metadata', 'version', version)
+    with open(setupcfg_path, 'wb') as configfile:
+        config.write(configfile)
```
Comment (humitos) [inline tasks.py]: Isn't this the other way around?

    configfile.write(config)

File: tasks.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+
+@task
+def release(ctx, version):
+    """
+    Tag release of Read the Docs.
+
+    Do this after prepare task and manual cleanup/commit
+    """
+    ctx.run('git checkout master && git tag {version}'.format(version=version))
```
Comment (humitos) [inline tasks.py]: `git push --tags` or ...

Maybe after the `prepare` we need to stage and commit the changes. Then, `release` and after that `push --tags` but I think it should be written somewhere.

File: tasks.py (hunk 2 above)
Comment (agjohnson) [reply tasks.py]: Hah, that would make sense though. :)

The ConfigParser API is goofy:
https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.write

File: tasks.py (hunk 3 above)
Comment (agjohnson) [reply tasks.py]: Ah true, that would be a good addition.

I'm not yet sure of this workflow. I like the idea of performing one large step, but we might want to double check on things in between version update and commit/push. We'll figure out better ways in the future.

I'll add in the git tag push though, that should definitely live here.

Comment (humitos) [reply tasks.py]: Yeah, the push maybe shouldn't be automatic be mentioned to the user to not forget to do it :)

**Pull request merged**